src/storage/user-data.ts

```diff
@@ -86,7 +87,17 @@ export async function saveUserData( data: UserData ): Promise< void > {
 	const filePath = getUserDataFilePath();
 
 	const asString = JSON.stringify( toDiskFormat( data ), null, 2 ) + '\n';
-	await fs.promises.writeFile( filePath, asString, 'utf-8' );
+	try {
+		await atomically.writeFile( filePath, asString, 'utf-8' );
+	} catch ( error ) {
+		// Fall back to FS function in case the writing fails with EXDEV error.
+		// This issue might happen on Windows when renaming a file.
+		// Reference: https://github.com/sindresorhus/electron-store/issues/106
+		// eslint-disable-next-line @typescript-eslint/no-explicit-any
+		if ( ( error as any )?.code === 'EXDEV' ) {
+			await fs.promises.writeFile( filePath, asString, 'utf-8' );
```

Comment:
I wonder if we should move this await fallback to a separate try/catch. If the regular writeFile fails it will crash the app. I guess that will be captured by Sentry anyway, but maybe we can fail gracefully.

Reply:
I think the failure should be handled by the caller. For instance, if there's a write error when creating a site, we should let the IPC handler catch the error and fail gracefully. WDYT?